MayssaraOmar/assignment3/Word search/src/test_dictionary.cpp

```diff
@@ -0,0 +1,93 @@
+#include "test_dictionary.h"
+#include "test.h"
+#include "exception"
+
+dictionary test_dictionary::dic;
+
+void test_dictionary::run_all_tests() {
+	insert_oneword();
+	insert_vector();
+	remove_oneword();
+	remove_vector();
+	isWord_true();
+	isWord_false();
+	isPrefix_true();
+	isPrefix_false();
+	clear_empty();
+	empty_dictionary();
+}
+void test_dictionary::insert_oneword() {
+	dic.insert("hello");
```

Comment:
It is often the case that in your test you want to check that things behaved correctly.
For example, here you could say:

```c++
   dic.insert("hello");
   EXPECT_TRUE(dic.isWord("hello"));
   EXPECT_TRUE(dic.isPrefix("h"));
   EXPECT_TRUE(dic.isPrefix("he"));
   EXPECT_TRUE(dic.isPrefix("hel"));
   EXPECT_TRUE(dic.isPrefix("hell"));
   EXPECT_TRUE(dic.isPrefix("hello"));
```

Otherwise your test runs the code but it does not check that things worked.
This shows how it is not always possible to test functions completely in isolation.